src/ragger/backend/speculos.py

```diff
@@ -159,22 +171,25 @@ def compare_screen_with_snapshot(self,
         else:
             return screenshot_equal(f"{golden_snap_path}", snap)
 
-    def wait_for_screen_change(self, timeout: float = 10.0, context: Any = None) -> Any:
-        start = time()
-        if not context:
-            return self._client.get_current_screen_content()
-        else:
-            content = self._client.get_current_screen_content()
-            while content == context:
-                # Give some time to other threads, and mostly Speculos one
-                sleep(0.2)
-                if (time() - start > timeout):
-                    raise TimeoutError("Timeout waiting for screen change")
-                content = self._client.get_current_screen_content()
-            return content
-
-    def compare_screen_with_text(self, text: str):
-        return text in dumps(self._client.get_current_screen_content())
-
-    def get_current_screen_content(self) -> Any:
+    def _get_current_screen_content(self) -> dict:
```

Comment:
I find having  `_get_current_screen_content` and  `get_current_screen_content` a bit confusing @xchapron-ledger

Reply:
@agrojean-ledger I've remove the dummy  `_get_current_screen_content` "helper".